backend/src/main/java/com/htecgroup/skynest/annotation/validator/UniqueCompanyValidator.java

```diff
@@ -0,0 +1,36 @@
+package com.htecgroup.skynest.annotation.validator;
+
+import com.htecgroup.skynest.annotation.UniqueCompany;
+import com.htecgroup.skynest.exception.company.PibAlreadyInUseException;
+import com.htecgroup.skynest.exception.register.EmailAlreadyInUseException;
+import com.htecgroup.skynest.exception.register.PhoneNumberAlreadyInUseException;
+import com.htecgroup.skynest.model.request.CompanyAddRequest;
+import com.htecgroup.skynest.repository.CompanyRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+@Component
+@AllArgsConstructor
+public class UniqueCompanyValidator
+    implements ConstraintValidator<UniqueCompany, CompanyAddRequest> {
+
+  private final CompanyRepository companyRepository;
+
```

Comment:
@petarstoimenovskihtec Is throwing exceptions here too bad a practice? I know I should be returning false, but that throws a ConstraintViolationException, which we are already handling as 403-FORBIDDEN.

Reply:
ConstraintViolationException should not be handled as 403 error, it is always 400 error, because it validates the business logic. If you are handling it as 403, then you have some other problem.
Validator should just return true/false in my opinion

Reply:
I don't like to block you now, but after merge, consider fixing it

Reply:
Makes sense, will change it when we fix ConstraintViolationException handling.